### PR TITLE
Avoid the use of abs64 in timedata

### LINKED ISSUE #20135: UBSan warning when fuzzing abs64(...)
When extending the `test/fuzz/integer` fuzzer I noticed the following UBSan warning when fuzzing `abs64(...)`:

```
runtime error: negation of -9223372036854775808 cannot be represented in type 'int64_t' (aka 'long'); cast to an unsigned type to negate this value to itself
```

Fuzzing harness:

```diff
diff --git a/src/test/fuzz/integer.cpp b/src/test/fuzz/integer.cpp
index 35d6804d4..bc158e5a2 100644
--- a/src/test/fuzz/integer.cpp
+++ b/src/test/fuzz/integer.cpp
@@ -40,6 +40,8 @@
 #include <set>
 #include <vector>

 void initialize()
 {
     SelectParams(CBaseChainParams::REGTEST);
@@ -82,6 +84,7 @@ void test_one_input(const std::vector<uint8_t>& buffer)
     (void)ComputeMerkleRoot(v256);
     (void)CountBits(u64);
     (void)DecompressAmount(u64);
+    (void)abs64(i64);
     (void)FormatISO8601Date(i64);
     (void)FormatISO8601DateTime(i64);
     // FormatMoney(i) not defined when i == std::numeric_limits<int64_t>::min()
```

Typically `abs(I n)` type functions are not defined when `n == std::numeric_limits<I>::min()` so it could be argued that this is expected, but perhaps the function could be rewritten in a way which guarantees that it gives the same behaviour across systems (instead of UB).

### PR DESCRIPTION
**Pull request by sipa**
Fixes #20135.

### CODE REVIEW

File: src/timedata.cpp [hunk 1]
```diff
@@ -79,7 +74,8 @@ void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
         int64_t nMedian = vTimeOffsets.median();
         std::vector<int64_t> vSorted = vTimeOffsets.sorted();
         // Only let other nodes change our time by so much
-        if (abs64(nMedian) <= std::max<int64_t>(0, gArgs.GetArg("-maxtimeadjustment", DEFAULT_MAX_TIME_ADJUSTMENT))) {
+        int64_t max_adjustment = std::max<int64_t>(0, gArgs.GetArg("-maxtimeadjustment", DEFAULT_MAX_TIME_ADJUSTMENT));
+        if (nMedian < -max_adjustment || nMedian > max_adjustment) {
```
Comment (kallewoof) [inline src/timedata.cpp:78]: I think this is flipped. Now if `nMedian` is beyond the max `nTimeOffset` is set to `nMedian`, whereas it was the opposite before:

Example: max = 4, nMedian = -3
Before: abs64(-3) <= 4 == 3 <= 4 == true
Now: -3 < -4 || -3 > 4 == false

Comment (sipa) [reply src/timedata.cpp]: Uh, fixed. Thanks!

**Comment by MarcoFalke** [comment]
ACK d1292f25f272401da0c58580521c74b1fa03a9ad 🎹

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK d1292f25f272401da0c58580521c74b1fa03a9ad 🎹
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhLDQwAtYxZPYoMjle0YqgTWENDhQOispBXpj557FjzlRPsr1Lbo2XIhyuqDtJr
vYrvpwVxPwjx/v/1kWqVtmiOhEPzkMwguDH6wVl3zMfnb97qzYE64p/JZp3AXLuD
1FOhGkoACVEzHMoHL25tzQfiH+AfJ44QD1iFn4KLnVk4YXmqqC5OB8J6Sdpm3beV
+2nHGmzSH+DM/IH09EP1gtfmBISVLlMJESc0Jf+mwdBhO/y99UK9NATfKo/Cw0i3
ewblc/Asehq8u7ZxGAMoHTlr1VgSTNdUZ0bdMKKigvRkfIoAg5kTq80/uKS7mg3T
InA6Lau/CWDlmB+I6opTWIZgCqt3ZUkixtAhWeC8RzSB+CNk0efkbv8YIC+jv/FJ
RtG6YMnSf8pfyZtvz8vpSOPl58U4jx2Z5D/+rD1mnxuc6d+gzAFJMqms3vwaOe3r
HxrdNIEGFnhaxqwd2wq/wA1cqCzsPIcJKmg1+q581YPkNCXlsaORvLl5FywpbBbg
54gywunH
=8Rrg
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9182aa2ebb693c9de1307af30edc9e597d470f51bd1cdb4b1d6b11923c3ef0db  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089182aa2ebb693c9de1307af30edc9e597d470f51bd1cdb4b1d6b11923c3ef0dbf01061b06dbd9d90cab1a69914b50905d65408fff0100086c7d894bf6e39d7e2218739faa0e408f1045f855849f0089f67c5a79ef9452f0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0104fb095fd5dac1bf488886914f63a357708f1045f85584bf00819e3277d16e8036b0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0107b5e322e8650ab2167031b4a525b75a008f1045f855849f0089e4984fbc37f0d440083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010a623e4fc48c7fa29285f90b973f7a54508f0204f7d14cb12826be5ec0aebae3906913bb47f622bdbc57095aa3b1360c9ab028a08f02097e8fedf1146d220513ba35856c9d2dae50f93675cdabdffd2f9d408ab06f8f708f1045f85584af008f441fa7385664a550083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

**Review by kallewoof — APPROVED** [review]
ACK d1292f25f272401da0c58580521c74b1fa03a9ad

Verified no behavioral change.

**Comment by practicalswift** [comment]
ACK d1292f25f272401da0c58580521c74b1fa03a9ad

**Review by jonatack — COMMENTED** [review]
ACK d1292f25f272401da0c58580521c74b1fa03a9ad code/logic review, verified there are no remaining callers of `abs64()`, verified no warnings in a debug build

**Pull request merged**